workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncher.cs

```diff
@@ -0,0 +1,334 @@
+
+                if (processResult.ExitCode != 0)
+                {
+                    return null;
+                }
+
+                // Get deployments from output.
+                var deploymentList = new List<DeploymentInfo>();
+                foreach (var line in processResult.Stdout)
+                {
+                    Debug.Log("Line: " + line);
+                    var tokens = line.Split(' ');
+                    if (tokens.Length != 2)
+                    {
+                        continue;
+                    }
+                    deploymentList.Add(new DeploymentInfo
+                    {
+                        Id = tokens[0],
+                        Name = tokens[1]
+                    });
+                }
+                return deploymentList;
+            }
+
+            private static string GenerateTempSnapshot()
+            {
+                string snapshotPath = FileUtil.GetUniqueTempPathInProject();
+
+                Snapshot snapshot = new Snapshot();
+                var simulatedPlayerCoordinatorTrigger = FpsEntityTemplates.SimulatedPlayerCoordinatorTrigger();
+                snapshot.AddEntity(simulatedPlayerCoordinatorTrigger);
+                snapshot.WriteToFile(snapshotPath);
+
+                return snapshotPath;
+            }
+
+            private static bool CheckDependencies()
+            {
+                var hasDotnet = !string.IsNullOrEmpty(Tools.Common.DotNetBinary);
+
+                if (hasDotnet)
+                {
+                    return true;
+                }
+
+                var builder = new StringBuilder();
+
+                builder.AppendLine(
+                    "The SpatialOS GDK for Unity requires 'dotnet' on your PATH to run its tooling.");
+                builder.AppendLine();
+
+                builder.AppendLine("Could not find 'dotnet' on your PATH.");
+
+                builder.AppendLine();
+                builder.AppendLine("If these exist on your PATH, restart Unity and Unity Hub.");
+                builder.AppendLine();
+                builder.AppendLine("Otherwise, install them by following our setup guide:");
+                builder.AppendLine("https://docs.improbable.io/unity/alpha/content/get-started/set-up");
+
+                EditorApplication.delayCall += () =>
+                {
+                    EditorUtility.DisplayDialog("GDK dependencies check failed", builder.ToString(), "OK");
+                };
+
+                return false;
+            }
+
+            private static string[] ConstructArguments(List<string> args)
+            {
+                var baseArgs = new List<string>
+                    {
+                        "run",
+                        "-p",
+                        $"\"{DotNetProjectPath}\"",
+                        "--",
+                    };
+                foreach (var arg in args)
```

Comment:
Condense to `baseArgs.AddRange(args.Select(arg => $"\"{arg}""));`

Reply:
In gamecode yeah, but here, a LINQ statement that's run in response to a button click will be fine.
Rule of thumb: if it's run every frame, definitely avoid, if it's in response to user input (e.g. at most a few times every few seconds) then it's fine.

Reply:
Done.